sound/soc/sof/control.c

```diff
@@ -342,21 +345,41 @@ int snd_sof_bytes_ext_get(struct snd_kcontrol *kcontrol,
 	struct snd_sof_dev *sdev = scontrol->sdev;
 	struct sof_ipc_ctrl_data *cdata = scontrol->control_data;
 	unsigned int tag = 0;
+	int max_length = SOF_IPC_MSG_MAX_SIZE -
+		sizeof(const struct sof_ipc_ctrl_data) -
+		sizeof(const struct sof_abi_hdr);
 
 	pm_runtime_get_sync(sdev->dev);
 
+	/* Decrement size for room for ext bytes header and find out
+	 * the limit from ext bytes size from topology and SOF IPC
+	 * size limit.
+	 */
+	size -= 2 * sizeof(u32);
+	if (max_length > be->max)
+		max_length = be->max;
+
+	dev_dbg(sdev->dev, "request size minus header is %d\n", size);
+	dev_dbg(sdev->dev, "max size is %d\n", max_length);
+
+	/* Limit size to prevent read of kernel data */
+	if (size > max_length) {
+		size = max_length;
+		dev_dbg(sdev->dev, "warning: the copy length was limited\n");
+	}
+
 	dev_dbg(sdev->dev, "getting data and command is %d\n", scontrol->cmd);
```

Comment:
can we please remove this line? this was added for debug purposes when working on the demo.

Reply:
Yep, sure.